File: java8-stream/src/main/java/com/bh/stream/data/Person.java
package com.bh.stream.data;

import lombok.Builder;
import lombok.Data;
import lombok.ToString;

@ToString
public class Person {
    private Integer id;
    private String name;
    private Integer age;
    private Integer score;

    public Person setId(Integer id) {
        this.id = id;
        return this;
    }

    public Person setName(String name) {
        this.name = name;
        return this;
    }

    public Person setAge(Integer age) {
        this.age = age;
        return this;
    }

    public Person setScore(Integer score) {
        this.score = score;
        return this;
    }

    public Integer getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public Integer getAge() {
        return age;
    }

    public Integer getScore() {
        return score;
    }
}

File: kotlin/kotlin-6/src/main/kotlin/com/bh/guava/Ordering/Person.kt
package com.bh.guava.Ordering

data class Person(
    val name: String,
    val age: Int
) : Comparable<Person> {
    override fun compareTo(other: Person): Int {
        // compareTo ：如果返回值是负数，表示比较的数小于other,如果返回值是正数,反之大于other，返回值是0，则比较的值等于other
        return age.compareTo(other.age)
    }
}
